extensions/wikia/Recirculation/RailContentService.php

```diff
@@ -2,9 +2,26 @@
 
 class RailContentService extends WikiaService {
 
+	const RAIL_THUMBNAIL_WIDTH = 94;
+	const RAIL_THUMBNAIL_HEIGHT = 53;
+
 	public function renderRailModule() {
 		$popularPagesService = new PopularPagesService();
-		$articles = $popularPagesService->getPopularPages( 5, 80, 50 );
+		$articles = $popularPagesService->getPopularPages(
+			5,
+			static::RAIL_THUMBNAIL_WIDTH,
+			static::RAIL_THUMBNAIL_HEIGHT
+		);
+
+		foreach ( $articles as $pageId => $article ) {
+			$gen = ( new VignetteUrlToUrlGenerator( $article['thumbnail'] ) )->build();
```

Comment:
you can use `VignetteRequest::fromUrl()` here. I'd also wrap it in `try...catch`
Like here: https://github.com/Wikia/app/blob/f74556613331e60006a4a7a49f6368657a8625f9/extensions/wikia/FeedsAndPosts/Thumbnails.class.php#L6